app/src/main/java/com/github/polypoly/app/utils/global/GlobalInstances.kt

```diff
@@ -34,12 +36,22 @@ class GlobalInstances {
             }
         }
 
-        var currentUser : User = User(7, "fake_user", "I am fake until google db is fully setup", Skin.default(),
-            Stats(0, 0, 0, 0, 0), listOf(), mutableListOf()
-        )
-        var currentFBUser : FirebaseUser? = null
+        // ============================================== CURRENT USER
+        var currentUser : User? = null
         var isSignedIn = false
 
+        fun initCurrentUser(key: String) {
+            remoteDB.keyExists<User>(key).thenAccept { exists ->
+                if(exists) {
+                    remoteDB.getValue<User>(key).thenAccept { currentUser = it }
+                } else {
+                    currentUser = User(id = key)
+                    remoteDB.registerValue(currentUser!!)
```

Comment:
I suggest to also initialize a simple profile for new users, using their names (that we can get from FBAuthUser). Or get them a random name, but anything to enable differenciation for players if they never change their name! :)

Reply:
Didn't know about the FBAuthUser name! I'll add it :)